### PR TITLE
Tasks and scripts to manage governor proposals

### PR DESCRIPTION
**Pull request by clemsos**
# Description

A set of hardhat tasks and scripts to manage/submit/debug GovernorAlpha proposals.

-- see the discussion in #7661 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/README.md [hunk 1]
```diff
@@ -172,7 +172,45 @@ Deploying Governor on localhost with the account: 0xf39Fd6e51aad88F6F4ce6aB88272
 > Governor deployed (w proxy) at: 0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9
 > Governor added to Timelock as sole proposer.  0xDc64a140Aa3E981100a9becA4E685f962f0cF6C9 is Proposer: true
 > Unlock Owner recounced Admin Role.  0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 isAdmin: false
-````
```
Comment (julien51) [inline smart-contracts/README.md:175]: Amazing! We should absolutely write a blog post about this BTW!

File: smart-contracts/scripts/deployments/governor.js [hunk 2]
```diff
@@ -30,8 +30,9 @@ async function main() {
     'UnlockProtocolTimelock'
   )
 
-  // one week in seconds
-  const MINDELAY = 60 * 24 * 7
```
Comment (julien51) [inline smart-contracts/scripts/deployments/governor.js:34]: poups! I forgot about that one/ We'll have to think about ways to tell everyone we fudged it.

File: smart-contracts/scripts/deployments/governor.js [hunk 3]
```diff
@@ -82,6 +83,15 @@ async function main() {
     ` (tx: ${governor.deployTransaction.hash})`
   )
 
+  if (networkName === 'localhost') {
+    // bring default voting period to 10 blocks while developing locally
+    await network.provider.send('hardhat_setStorageAt', [
```
Comment (julien51) [inline smart-contracts/scripts/deployments/governor.js:88]: Wow, you will have to tell me more about this blackmagic

**Review by julien51 — APPROVED** [review]
💯

File: smart-contracts/test/helpers/mainnet.js [hunk 4]
```diff
@@ -28,9 +29,47 @@ const impersonate = async (address) => {
...
+  const index = ethers.utils.solidityKeccak256(
+    ['uint256', 'uint256'],
+    [recipientAddress, 51] // key, slot
+  )
+
+  // Manipulate local balance (needs to be bytes32 string)
+  await network.provider.request({
+    method: 'hardhat_setStorageAt',
+    params: [udtAddress, index.toString(), toBytes32(udtAmount).toString()],
+  })
```
Comment (clemsos) [inline smart-contracts/test/helpers/mainnet.js:64]: This helper to give UDT works but it is a bit of a dirty hack that modify directly the contract storage. Maybe it is better to use a more proper way, like impersonate Unlock, then mint and transfer tokens.

Comment (clemsos) [reply smart-contracts/test/helpers/mainnet.js:64]: For instance it will make `totalSupply` wrong

**Comment by julien51** [comment]
Ok I think it's still good! Thanks!

**Pull request merged**